app/src/org/commcare/android/logging/ForceCloseLogger.java

```diff
@@ -112,12 +117,31 @@ public String getFilename() {
         }
 
         try {
-            HttpResponse response = generator.postData(URI, entity);
+            HttpResponse response = generator.postData(submissionUri, entity);
             ByteArrayOutputStream bos = new ByteArrayOutputStream();
             response.getEntity().writeTo(bos);
             Log.d(TAG, "Response: " + new String(bos.toByteArray()));
         } catch (IOException e) {
             e.printStackTrace();
+            return false;
+        }
+
+        return true;
+    }
+
+    private static void writeErrorToStorage(ForceCloseLogEntry entry) {
+        if (logStorage != null) {
+            logStorage.write(entry);
+        }
+    }
+
+    private static String getSubmissionUri() {
+        try {
+            SharedPreferences settings = CommCareApplication._().getCurrentApp().getAppPreferences();
+            return settings.getString(CommCarePreferences.PREFS_SUBMISSION_URL_KEY,
+                    CommCareApplication._().getString(R.string.PostURL));
+        } catch (Exception e) {
+            return CommCareApplication._().getString(R.string.PostURL);
```

Comment:
I don't know for sure what it's trying to catch here, this was just refactored from this: https://github.com/dimagi/commcare-odk/blob/master/app/src/org/commcare/tasks/ExceptionReporting.java#L74-L80. I can't really imagine what else it would be protecting against besides the current app being null though, so feel comfortable changing it to that.